### DISCUSSION TITLE
I don't understand how x-ref has changed in v3. Please help.

### DISCUSSION THREAD

**Discussion by tanthammar** [alpinejs/alpine#1952] [1. Help] [answered]
In Alpine v2 I had this, without any issues.
```blade
<div x-data @focus-input.window="$refs.searchInput.focus()">
    <input x-ref="searchInput" />
</div>
```
In Livewire I emitted
```php
$this->dispatchBrowserEvent("focus-input");
```
... and the input would focus.

In Alpine v3 I get an error, and the field doesn't get focused.
```js
Uncaught TypeError: Cannot read property '0' of undefined
```
The error points me to this line in Alpine js
```js
getAlpineDataInstance(node) {
 if (this.isV3) {
   return node._x_dataStack[0]
 }
   return node.__x
 }
```

I had to change to vanilla js 
```blade
<div x-data="{
        focusInput() {
            document.getElementById({{$dynamicID}}').focus();
        }
     }"
   @focus-input.window="focusInput()"
>
<input id="{{$dynamicID}}" />
</div>
```

Can someone please help me understand what has changed.

**Comment by SimoTod** [comment, marked as answer]
That error seems to be coming from alpine dev tools, not Alpine. Does it work if you remove the dev tool? It's probably worth to open an issue or a discussion there.

**Reply by tanthammar** [reply]
Thank you @SimoTod , yes it was from Alpine devtools.
https://github.com/alpine-collective/alpinejs-devtools/issues/295
